libs/types/src/election_adt.ts

```diff
@@ -0,0 +1,257 @@
+} from './election';
+
+export function findById<T extends { id: string }>(
+  array: readonly T[],
+  id: string
+): T {
+  return find(array, (element) => element.id === id);
+}
+
+export interface State {
+  name(): string;
+}
+
+export interface County {
+  name(): string;
+}
+
+export interface District {
+  name(): string;
+}
+
+export interface Precinct {
+  readonly id: string;
+  name(): string;
+}
+
+export interface Party {
+  name(): string;
+  ballotLabel(): string;
+  abbreviation(): string;
+}
+
+type ContestType = 'candidate' | 'ballot-measure';
+
+interface BaseContest {
+  readonly type: ContestType;
+  readonly id: string;
+  title(): string;
+  party(): Party | undefined;
+}
+
+export interface CandidateContest extends BaseContest {
+  readonly type: 'candidate';
+  votesAllowed(): number; // Replaces seats
+  candidates(): readonly Candidate[];
+  allowWriteIns(): boolean;
+}
+
+export interface Candidate {
+  readonly id: string;
+  name(): string;
+  party(): Party | undefined;
+}
+
+export interface BallotMeasureContest extends BaseContest {
+  readonly type: 'ballot-measure';
+  description(): string;
+  yesOption(): BallotMeasureOption;
+  noOption(): BallotMeasureOption;
+}
+
+export interface BallotMeasureOption {
+  readonly id: string;
+  label(): string;
+}
+
+export type Contest = CandidateContest | BallotMeasureContest;
+
+export interface BallotStyle {
+  readonly id: string;
+  precincts(): readonly Precinct[];
+  contests(): readonly Contest[];
+  party(): Party | undefined;
+}
+
+export interface Election {
+  // Logical election
+  title(): string;
+  date(): string;
+  state(): State;
+  county(): County;
+  districts(): readonly District[];
+  precincts(): readonly Precinct[];
+  parties(): readonly Party[];
+  contests(): readonly Contest[];
+  ballotStyles(): readonly BallotStyle[];
+
+  // The rest of the fields below are the same as in the original VXF election
+  // definition data structure. They are deprecated, since we will be moving
+  // these features to CDF in the future. For now, this data will be appended as
+  // an extra VX-specific blob at the end of the CDF. We don't both making
+  // accessors for them now, since they will change.
+
+  // Physical ballot
+  /** @deprecated */
+  ballotLayout?: BallotLayout;
+  /** @deprecated */
+  gridLayouts?: readonly GridLayout[];
+  /** @deprecated */
+  seal?: string;
+  /** @deprecated */
+  sealUrl?: string;
+  /** @deprecated */
+  quickResultsReportingUrl?: string;
+
+  // Election settings
+  /** @deprecated */
+  markThresholds?: MarkThresholds;
+  /** @deprecated */
+  centralScanAdjudicationReasons?: readonly AdjudicationReason[];
+  /** @deprecated */
+  precinctScanAdjudicationReasons?: readonly AdjudicationReason[];
+
+  // Localization
+  /** @deprecated */
+  _lang?: Translations;
+  /** @deprecated */
+  ballotStrings?: BallotStrings;
+}
+
+export function buildElectionFromVxf(electionData: VxElectionData): Election {
```

Comment:
We (currently) call `electionData` the raw JSON string. Do you plan to rename that to avoid ambiguity?